Makefile

```diff
@@ -3,10 +3,24 @@
 # remember the right command. So let's keep this simple, just targets and
 # commands.
 
+# Ensure bash shell! Needed for checking go version...
+SHELL := /bin/bash
+
 # Ensure go-extpoints and go-import-subtree are available for go generate
 export PATH := $(GOPATH)/bin:$(PATH)
 
-all: build
+# For checking go compiler has a suitable version number
+GO_VERSION := $(shell go version 2>/dev/null | cut -f3 -d' ')
+GO_MAJ := $(shell echo "$(GO_VERSION)" | cut -f1 -d'.')
+GO_MIN := $(shell echo "$(GO_VERSION)" | cut -f2 -d'.')
+
+all: rebuild
+
+prechecks:
+	@test -n "$(GO_VERSION)" || (echo "Could not find go compiler, 'go version' returned no output" && false)
+	@test "$(GO_MAJ)" == "go1" || (echo "Require go version 1.x, where x >= 7; however found '$(GO_VERSION)'" && false)
+	@test "$(GO_MIN)" -ge 7 || (echo "Require go version 1.x, where x>=7; however found '$(GO_VERSION)'" && false)
```

Comment:
both `"0$(GO_MIN)"` and `"$(GO_MIN)"` will result in a falsy value for boolean logic:
```
jhford:~/taskcluster/sqs-simple $ test "" -gt 7
bash: test: : integer expression expected
jhford:~/taskcluster/sqs-simple $ echo $?
2
```
and adding the 0 doesn't account for the case that `GO_MIN` is assigned to some non-number, non-empty string value.
```
jhford:~/taskcluster/sqs-simple $ test "0john" -gt 7
bash: test: 0john: integer expression expected
jhford:~/taskcluster/sqs-simple $ echo $?
2
```
I think it reads cleaner without the 0, but no strong preference.